server/src/tree-sitter/analyzer.ts

```diff
@@ -828,43 +828,15 @@ export default class Analyzer {
       return undefined
     }
 
-    const scanResultDoc = lines.slice(index).join('\n')
-    const scanResultDocUri = originalDocUri + '.scanned'
-    const scanResultDocTree = this.parser.parse(scanResultDoc)
-
-    const scanResultDocGlobalDeclarations = getGlobalDeclarations({ tree: scanResultDocTree, uri: scanResultDocUri, getFinalValue: true })
-    const scanResultDocSymbols = this.getAllSymbolsFromGlobalDeclarations(scanResultDocGlobalDeclarations)
-    const originalDocDeclarationSymbols = this.getAllSymbolsFromGlobalDeclarations(analyzedOriginalDoc.globalDeclarations)
-    const originalDocVariableExpansionSymbols = analyzedOriginalDoc.variableExpansionSymbols
-
-    const scannedResultSymbolInfo: BitbakeSymbolInformation[] = []
-    // Store the variable expansion symbols first and use their final value to resolve those declaration symbols. e.g. VAR:${PN}
-    originalDocVariableExpansionSymbols.forEach((symbol) => {
-      const foundSymbol = scanResultDocSymbols.find((scanResultDocSymbol) => {
-        return this.symbolsAreTheSame(scanResultDocSymbol, symbol)
-      })
-      if (foundSymbol !== undefined) {
-        scannedResultSymbolInfo.push(foundSymbol)
-      }
-    })
-
-    originalDocDeclarationSymbols.forEach((symbol) => {
-      const resolvedSymbol = this.resolveSymbol(symbol, scannedResultSymbolInfo)
+    const scanResultText = lines.slice(index).join('\n')
+    const scanResultDummyUri = originalDocUri + '.scanned'
```

Comment:
We're going to need the symbols not for just the URI, but all files related to that recipe (.inc, .bb, .bbappend).
I don't think you need to store the text as well. It assumes we're going to parse it again which we should not.
I expect the data to be stored to be <String, scanResultParsedTree>.
String should be the name of a recipe. "busybox" for instance. When analyzing a document, the analyzer should try to find within the symbols data base if we have a record corresponding.

The record should contain all variables final values + their assignment history. The analyzer will use them to provide hover final values and accurate definitions if the current document corresponds to the recipe name.

(Minor) On close of a document matching, we should check no other remaining open document corresponds to that recipe before removing the symbols.

Reply:
Check the newly added commits for 2).

For 4), only the client side has access to the opened editors. This means I need to send an additional custom onClose request to the server and pass the opened editors in the param. I could make use of the onDidOpen to keep track of opened files on the server side, but it is tedious.

I think we should remove it on another event other than closing a file or we could have an extra command to do that or we just don't remove it at all.

Personally, I think it is ok that we don't remove it.

Reply:
Keeping the scan in memory is a bonus however we must free memory regularly in order to stay lean. The user can always redo a scan easily.
The simplest tradeoff is to delete the scan from memory when any matching .bb/.inc/.bbappend is closed. If you have an easy tradeoff to improve, that's better but not critical.

Reply:
ok, then it is best to go for the request method. Will add it later.